operator/apis/loki/v1/lokistack_types.go

```diff
@@ -791,6 +791,112 @@ type IngestionLimitSpec struct {
+type OTLPAttributesSpec struct {
+	// Action defines the indexing action for the selected attributes.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Action"
+	Action OTLPAttributeAction `json:"action,omitempty"`
+
+	// Attributes is the list of attributes to configure indexing or drop them
+	// altogether.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Attributes"
+	Attributes []string `json:"attributes,omitempty"`
+
+	// Regex to choose attributes to configure indexing or drop them
+	// altogether.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Regex"
+	Regex string `json:"regex,omitempty"`
+}
+
+// OTLPResourceAttributesSpec contains the configuration for resource attributes
+// to store them as index labels or Structured Metadata or drop them altogether.
+type OTLPResourceAttributesSpec struct {
+	// IgnoreDefaults whether to ignore the default global list of resource attributes
+	// indexed as labels.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch",displayName="Ignore Global Defaults"
+	IgnoreDefaults bool `json:"ignoreDefaults,omitempty"`
+
+	// Attributes contains the configuration for resource attributes
+	// to store them as index labels or Structured Metadata or drop them altogether.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Attributes"
+	Attributes []OTLPAttributesSpec `json:"attributes,omitempty"`
+}
+
+// GlobalOTLPSpec defines which resource, scope and log attributes to
+// be stored as index or structured metadata or drop altogether for all
+// tenants.
+type GlobalOTLPSpec struct {
+	// IndexedResourceAttributes contains the global configuration for resource attributes
+	// to store them as index labels or Structured Metadata or drop them altogether.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Action"
+	IndexedResourceAttributes []string `json:"indexedResourceAttributes,omitempty"`
+
+	OTLPSpec `json:",omitempty"`
+}
+
+// OTLPSpec defines which resource, scope and log attributes to
+// be stored as index or structured metadata or drop altogether
+type OTLPSpec struct {
+	// ResourceAttributes contains the configuration for resource attributes
+	// to store them as index labels or Structured Metadata or drop them altogether.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resource Attributes"
+	ResourceAttributes OTLPResourceAttributesSpec `json:"resourceAttributes,omitempty"`
```

Comment:
Make it a pointer, if it's optional?

```suggestion
	ResourceAttributes *OTLPResourceAttributesSpec `json:"resourceAttributes,omitempty"`
```